### ISSUE TITLE
KV cache reuse: Decode Error -1 (n_tokens == 0) on iteration 2 with Qwen 3.6

### ISSUE THREAD

**Issue by fpt** [fpt/rs-gallium#98]
## What was tried

`unsloth/Qwen3.6-35B-A3B-GGUF` (UD-Q3_K_XL) via `configs/qwen3.6.toml`, running the `file_read` testsuite case (read a file, report its contents — a two-iteration ReAct turn: iteration 1 calls `Read`, iteration 2 should answer from the tool result). Reproduced with the default KV cache reuse settings (`GALLIUM_KV_CACHE_SLOTS` unset, so reuse on, 1 slot).

## Symptom

Iteration 2 fails immediately, before any generation:

```
Error: Network error: Initial decode failed: Decode Error -1: n_tokens == 0
```

Reproduced identically at both `GALLIUM_GPU_LAYERS=26` and at full offload (999) with the new `GALLIUM_CPU_MOE=1` (#[PR link once opened]) — same error either way, so it is **not** a GPU-layers/VRAM problem (ruled out explicitly: this is the same class of failure this session spent a lot of time on for Gemma 4/#92, and it isn't that — no "Failed to create context", no context-creation retry loop, and it reproduces with VRAM headroom to spare under `cpu_moe`).

`GALLIUM_KV_CACHE_SLOTS=0` (reuse off) avoids it entirely — `file_read` then passes cleanly (`✓ Found 'ZUCCHINI' (read tool worked)`).

## Why this looks like a real bug, not a config problem

CLAUDE.md's KV cache reuse section documents an explicit invariant this mechanism is supposed to guarantee:

> **One token is always left to evaluate.** The sampler reads the logits of the last position *evaluated*; a fully-cached prompt evaluates nothing and would sample from whatever the previous call left behind.

`Decode Error -1: n_tokens == 0` is llama.cpp refusing a decode call with an empty batch — which is exactly the case that invariant exists to prevent. Something in `llm_local.rs`'s slot/suffix-diffing logic is computing zero new tokens to evaluate for iteration 2's prompt against Qwen 3.6's chat template rendering, rather than the "at least one" the design calls for.

Not investigated further this session (scope was a `--n-cpu-moe`-style feature try, and this surfaced along the way, unrelated to it): the specific reason the diff comes out to zero for this model/prompt shape, and whether it's Qwen-3.6-template-specific or a more general edge case (a same-generation two-iteration `file_read` turn on `gemma4-26b-cuda-12gb` passed cleanly earlier this session with default reuse settings, so it isn't universal — but that doesn't rule out other models/templates hitting it too).

## Reference

- Workaround: `GALLIUM_KV_CACHE_SLOTS=0` for this model until fixed.
- Relevant code: `llm_local.rs`'s slot pool / suffix-evaluation logic (see CLAUDE.md's "KV cache reuse across ReAct iterations" section for the intended invariants).
